diff --git a/Makefile b/Makefile
index 7d4254d..9981d60 100644
--- a/Makefile
+++ b/Makefile
@@ -62,13 +62,14 @@ endif
 #
 
 FONTDUMP_EXE := $(OBJDIR)/fontdump
+CMAPDUMP_EXE := $(OBJDIR)/cmapdump
+ifneq "$(OS)" "Rtems"
 $(FONTDUMP_EXE): $(OBJDIR)/fontdump.o
 	$(LD_CMD)
 
-CMAPDUMP_EXE := $(OBJDIR)/cmapdump
 $(CMAPDUMP_EXE): $(OBJDIR)/cmapdump.o
 	$(LD_CMD)
-
+endif
 #
 # Sources
 #
@@ -214,9 +215,8 @@ $(GENDIR)/font_cjk.c: $(FONTDUMP_EXE) $(CJKFONT_FILES)
 
 endif
 
-FONT_SRC := \
-	$(GENDIR)/font_base14.c \
-	$(GENDIR)/font_cjk.c
+FONT_SRC := $(GENDIR)/font_base14.c
+FONT_SRC := $(FONT_SRC) $(GENDIR)/font_cjk.c
 
 FONT_OBJ := $(FONT_SRC:$(GENDIR)/%.c=$(OBJDIR)/%.o)
 
@@ -281,9 +281,8 @@ $(GENDIR)/cmap_korea.c: $(CMAPDUMP_EXE) $(CMAP_KOREA_FILES)
 
 endif
 
-CMAP_SRC := \
-	$(GENDIR)/cmap_unicode.c \
-	$(GENDIR)/cmap_cns.c \
+CMAP_SRC := $(GENDIR)/cmap_unicode.c
+CMAP_SRC := $(CMAP_SRC) $(GENDIR)/cmap_cns.c \
 	$(GENDIR)/cmap_gb.c \
 	$(GENDIR)/cmap_japan.c \
 	$(GENDIR)/cmap_korea.c
diff --git a/Makerules b/Makerules
index 01c45df..8fc1e7d 100644
--- a/Makerules
+++ b/Makerules
@@ -4,8 +4,8 @@ OS ?= $(shell uname)
 OS := $(OS:MINGW%=MINGW)
 
 CC ?= cc
-CFLAGS := -Ifitz -Imupdf -Wall
-LDFLAGS :=
+CFLAGS := $(CFLAGS) -Ifitz -Imupdf -Wall
+LDFLAGS := $(LDFLAGS)
 
 ifeq "$(build)" "debug"
 CFLAGS += -pipe -g
